palindrome_finder.py

```
@@ -1,19 +1,33 @@
+            for line in f:
+                for word in line.split():
+                    clean = clean_word(word.lower())
+                    if len(clean) > 1 and is_palindrome(clean):
+                        palindromes.add(clean)
+    except FileNotFoundError:
+        print(f"Error: File '{filename}' not found.")
+        return
+    except PermissionError:
+        print(f"Error: Permission denied to read file '{filename}'.")
+        return
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)